internal/storage/field_stats.go

```diff
@@ -141,12 +144,22 @@ func (stats *FieldStats) UnmarshalJSON(data []byte) error {
 			}
 		}
 
-		if bfMessage, ok := messageMap["bf"]; ok && bfMessage != nil {
-			stats.BF = &bloom.BloomFilter{}
-			err = stats.BF.UnmarshalJSON(*bfMessage)
+		bfType := bloomfilter.BasicBF
+		if bfTypeMessage, ok := messageMap["bfType"]; ok && bfTypeMessage != nil {
+			err := json.Unmarshal(*bfTypeMessage, &bfType)
 			if err != nil {
 				return err
 			}
+			stats.BFType = bfType
```

Comment:
do we need to set stats.BFType to a default value?

Reply:
in marshal impl, stats.BFType has been already set to current default bf type